### PR TITLE
NUI-1679 add e2e blocktransfer test

### PR DESCRIPTION
**Pull request by jdelbick**
## Description

Fixes # [(issue)](https://jira.corp.adobe.com/browse/NUI-1679)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: e2e/blocktransfer.test.js [hunk 1]
```diff
@@ -90,13 +110,26 @@ describe('Block Transfer e2e test', function() {
...
-        return doBlockDownload(fileUrl, fileSize, downloadFile);
+        await doBlockUpload(uploadFileUrls, fileSize, filePath);
+        console.log("** DONE UPLOADING");
+        // singular url used for downloading
+        const downloadFileUrl = getBlobUrl(filePath, {
+            permissions: "r",
+            size: fileSize
+        });
+        console.log("**** DOWNLOAD URL", downloadFileUrl);
+        return doBlockDownload(downloadFileUrl, fileSize, downloadFile);
```
Comment (jdelbick) [inline e2e/blocktransfer.test.js:133]: download is failing currently:
```
{"transfer_error":{"asset":{},"error":{}}}
    1) AEM upload then download


  0 passing (1s)
  1 failing

  1) Block Transfer e2e test
       AEM upload then download:
     TypeError: Cannot read property 'emit' of undefined
      at TransferController.<anonymous> (lib/block/blockdownload.js:78:18)
      at TransferController.notifyError (lib/controller/transfercontroller.js:180:14)
      at GetAssetMetadata.execute (lib/functions/getassetmetadata.js:100:28)
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

File: e2e/.env_example [hunk 2]
```diff
@@ -24,4 +24,9 @@ AEM_ENDPOINT=""
 # when it expires.
 #
 # EXAMPLE: login-token=1234567
-#LOGIN_TOKEN
\ No newline at end of file
+#LOGIN_TOKEN
```
Comment (tmathern) [inline e2e/.env_example:27]:
```suggestion
# LOGIN_TOKEN
```

**Review by tmathern — APPROVED** [review]

File: e2e/blocktransfer.test.js [hunk 3]
```diff
@@ -38,7 +39,7 @@ const {
 
 describe('Block Transfer e2e test', function() {
     this.timeout(60000);
-    async function doBlockUpload(fileUrl, fileSize) {
+    async function doBlockUpload(fileUrl, fileSize, filePath) {
```
Comment (pheenomenon) [inline e2e/blocktransfer.test.js:42]: is `filePath` not used in this function ? its [hardcoded at line 62](https://github.com/adobe/node-httptransfer/pull/83/files#diff-7723a12188d79813c09f606e9871007fab2542abd0d97de9527756db5bf7164eR62)

**Review by pheenomenon — APPROVED** [review]

File: e2e/blocktransfer.test.js (hunk 3 above)
Comment (jdelbick) [reply e2e/blocktransfer.test.js:42]: yes it is hardcoded for the test.
it is used for committing the blocks: https://github.com/adobe/node-httptransfer/blob/5405cfe7e2385acc0b87685e61265a4dc217c0ea/e2e/blocktransfer.test.js#L75

**Pull request merged**